### PR TITLE
Remove torch 1.13

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Remove torch 1.13

Along for ride, removes local checkpoints which has been killed by torch.

### CODE REVIEW

File: tests/utils/test_inference.py [hunk 1]
```diff
@@ -212,11 +212,10 @@ def test_export_for_inference_onnx(model_cls, sample_input, onnx_opset_version,
     if onnx_opset_version == None and version.parse(torch.__version__) < version.parse('1.13'):
         pytest.skip("Don't test prior PyTorch version's default Opset version.")
 
-    from composer.utils.misc import using_torch_2
-    if using_torch_2():
-        pytest.xfail(
-            'torch.onnx.errors.UnsupportedOperatorError: Exporting the operator "aten::unflatten" to ONNX opset version 14 is not supported.'
-        )
+    # TODO: Fix this test
```
Comment (dakinggg) [inline tests/utils/test_inference.py:215]: Jira instead of pr Todo please

File: composer/trainer/dist_strategy.py [hunk 2]
```diff
@@ -261,21 +249,13 @@ def sync_hook(*args):
...
-        # Setting it to `True` is mandatory when using `torch.compile()`.
-        kwargs['use_orig_params'] = fsdp_config['use_orig_params']
-        if version.parse(torch.__version__.split('.dev')[0]) >= version.parse('2.2.0'):
-            if 'device_mesh' in fsdp_config:
-                from torch.distributed._tensor import init_device_mesh
-                kwargs['device_mesh'] = init_device_mesh(
-                    'cuda',
-                    tuple([int(x) for x in fsdp_config['device_mesh']]),
-                )
+    if version.parse(torch.__version__.split('.dev')[0]) >= version.parse('2.2.0'):
```
Comment (dakinggg) [inline composer/trainer/dist_strategy.py:252]: Can we do this everywhere instead of the 2.2.9 thing?

File: composer/core/state.py [hunk 3]
```diff
@@ -532,11 +471,13 @@ def __init__(
...
 
-        if using_torch_2() and self.fsdp_state_dict_type == 'local':
-            raise DeprecationWarning(
-                textwrap.dedent(
-                    "FSDP state_dict_type='local' is deprecated in torch>=2.0.0. "
-                    "Please set fsdp_config['state_dict_type']='sharded' instead and will be removed in v0.17"))
+        if self.fsdp_state_dict_type not in [None, 'full', 'sharded']:
+            if self.fsdp_state_dict_type == 'local':
+                raise ValueError('Composer and PyTorch no longer support saving or loading local state dicts. '
+                                 'To upgrade an older checkpoint, use Composer version 0.18.1 and export to '
```
Comment (dakinggg) [inline composer/core/state.py:477]: "to as" -> "as"

File: .github/workflows/pr-cpu.yaml [hunk 4]
```diff
@@ -13,36 +13,26 @@ jobs:
...
-          container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
-          markers: "not daily and not remote and not gpu and vision and not doctest"
-          pytest_command: "coverage run -m pytest"
-          composer_package_name: "mosaicml"
-        - name: "cpu-doctest"
+        #   markers: not daily and not remote and not gpu and not vision and not doctest
+        #   pytest_command: coverage run -m pytest
+        #   composer_package_name: mosaicml
+        - name: cpu-doctest
           container: mosaicml/pytorch:1.13.1_cpu-python3.10-ubuntu20.04
```
Comment (dakinggg) [inline .github/workflows/pr-cpu.yaml:32]: This container should change?

File: .github/workflows/pr-cpu.yaml [hunk 5]
```diff
@@ -13,36 +13,26 @@ jobs:
...
-        - name: "cpu-3.10-2.1"
+          markers: not daily and not remote and not gpu and not vision and not doctest
+          pytest_command: coverage run -m pytest
+          composer_package_name: mosaicml
+        - name: cpu-3.10-2.1
           container: mosaicml/pytorch:2.1.0_cpu-python3.10-ubuntu20.04
-          markers: "not daily and not remote and not gpu and not vision and not doctest"
-          pytest_command: "coverage run -m pytest"
-          composer_package_name: "mosaicml"
-        # - name: 'cpu-3.10-2.2'
```
Comment (dakinggg) [inline .github/workflows/pr-cpu.yaml:31]: 2.2 should be active now?

File: .github/workflows/daily.yaml [hunk 6]
```diff
@@ -18,66 +18,56 @@ jobs:
...
           container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
-          markers: "not daily and (remote or not remote) and not gpu and vision and not doctest"
-          pytest_command: "coverage run -m pytest"
-          composer_package_name: "mosaicml"
-        - name: "cpu-doctest"
+          markers: not daily and (remote or not remote) and not gpu and vision and not doctest
+          pytest_command: coverage run -m pytest
+          composer_package_name: mosaicml
+        - name: cpu-doctest
           container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
```
Comment (dakinggg) [inline .github/workflows/daily.yaml:42]: Remove 1.13 image?

File: .github/workflows/daily.yaml [hunk 7]
```diff
@@ -18,66 +18,56 @@ jobs:
...
           container: mosaicml/pytorch:2.1.0_cpu-python3.10-ubuntu20.04
-          markers: "daily and (remote or not remote) and not gpu and not vision and not doctest"
-          pytest_command: "coverage run -m pytest"
-          composer_package_name: "composer"
-        - name: "daily-cpu-vision"
+          markers: daily and (remote or not remote) and not gpu and not vision and not doctest
+          pytest_command: coverage run -m pytest
+          composer_package_name: composer
+        - name: daily-cpu-vision
           container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
```
Comment (dakinggg) [inline .github/workflows/daily.yaml:62]: Remove 1.13?

File: .github/workflows/daily.yaml [hunk 8]
```diff
@@ -18,66 +18,56 @@ jobs:
...
-          markers: "daily and (remote or not remote) and not gpu and vision and not doctest"
-          pytest_command: "coverage run -m pytest"
-          composer_package_name: "mosaicml"
-        - name: "daily-cpu-doctest"
+          markers: daily and (remote or not remote) and not gpu and vision and not doctest
+          pytest_command: coverage run -m pytest
+          composer_package_name: mosaicml
+        - name: daily-cpu-doctest
           container: mosaicml/pytorch_vision:1.13.1_cpu-python3.10-ubuntu20.04
```
Comment (dakinggg) [inline .github/workflows/daily.yaml:67]: 1.13?

**Review by dakinggg — APPROVED** [review]
Should we deprecate orig params false too?

File: .github/workflows/pr-cpu.yaml (hunk 5 above)
Comment (mvpatel2000) [reply .github/workflows/pr-cpu.yaml:31]: Chuck will update in a separate PR where he bumps images

File: composer/trainer/dist_strategy.py (hunk 2 above)
Comment (mvpatel2000) [reply composer/trainer/dist_strategy.py:252]: Let's clean up in a separate PR? I think it's a different issue (but agree)

Comment (dakinggg) [reply composer/trainer/dist_strategy.py:252]: sure, jira plz

**Pull request merged**